platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -1298,6 +1299,15 @@ void OfflineDatabase::markUsedResources(int64_t regionID, const std::list<Resour
     handleError(ex, "mark resources as used");
 }
 
+std::exception_ptr OfflineDatabase::pack() try {
+    if (!db) initialize();
+    vacuum();
+    return nullptr;
+} catch (const mapbox::sqlite::Exception& ex) {
```

Comment:
catch generic exception? Initialize can throw IOException

Reply:
nice catch! seems it's missing across the storage, I'll make a separate pr to handle it.